### DISCUSSION TITLE
Scroll Display

### DISCUSSION THREAD

**Discussion by kfishermail** [j-kaltes/Juggluco#45] [General]
How can I configure Juggluco to display the latest data on the graph
and display current glucose on right hand side?
Right now Jugg updates the graph but keeps the starting time
constant.  After a while Jug displays more data on the graph, and the current
glucose value disappears (get overwritten)

Use case: I would like to mirror a display to an old android phone,
and have it always show the current glucose, and last 4 hours of
data.  Right now, I have to manually swipe the display every now and then
to make the current glucose visible

**Comment by j-kaltes** [comment]
Yes, the time period displayed should not move without user interaction. This will result in the current glucose value moving to the right with time. This can have the effect that the current glucose values shifts off the screen when you can't touch the phone for hours. You can remedy this a little bit by moving the current glucose value to the left and increasing the amount of time displayed on the screen. For example when you need to work four hours before you can wash your hands, you can adjust the amount of time displayed on the screen so that the width of four hours on the screen is equal to the length between the right end of the current glucose value and the right of the screen. See picture:
![moves](https://github.com/j-kaltes/Juggluco/assets/72013079/1f6fda29-7b80-4efa-9a4b-9f0a0fa8891e)
Between 10:00 and 14:00 is as much space as between 4.4 and the right border of the screen.

An alternative is to turn on Floating glucose or edit the source.

**Comment by AgushkaNightscout** [comment]
I upgraded my old laptop to make it a permanent working glucose monitor for my daughter. I put it higher up on a cupboard so I can see it from everywhere but I still have to climb up to it to move the Juggluco screen)) 

![WhatsApp Image 2024-02-14 at 20 55 281](https://github.com/j-kaltes/Juggluco/assets/90107639/38095c8e-6c67-433d-9d61-46959598fb80)

For young children it is important to see a trend, so I need an up-to-date minute-by-minute graph. I can, of course, create a second Nightscout, send the minute-by-minute data from Juggluco to it, and then switch on the split feature, but I like the Juggluco screen so much I don't want to do that!

![Снимок экрана от 2024-02-14 22-04-46](https://github.com/j-kaltes/Juggluco/assets/90107639/64fb1aeb-078d-4e1b-acbf-c04c841598ba)

Could you please tell me, Jaap, are you still not planning to implement auto-scroll?

**Comment by kfishermail** [comment]
I ultimately solved this by installing a program called "Macrodroid", and I
send
a simulated screen swipe every 20 seconds to my phone.

**Reply by AgushkaNightscout** [reply]
It's not easy in emulated Android, I'm afraid it's not even possible, it didn't work for me at least.

**Comment by kfishermail** [comment]
On another machine, I do a similar thing by connecting the phone to a PC
via USB, and then send

ADB commands to emulate a screen swipe.  Did a quick google search, it
appears ADB does work on
emulated android

**Reply by AgushkaNightscout** [reply]
Yes, even on an Android device, Macrdroid can't do "click" or "touch" action without root permissions and the Autoinput addon (originally for Tasker). I was able to install Macrodroid (x86 version) on the emulator, give it root permissions, but it still couldn't capture the click on date in the right Juggluco menu. Autoinput plugin exists only in arm version, it can't be installed on emulator. Looking for similar software for Lubuntu at the moment, this is probably the best way.

**Reply by AgushkaNightscout** [reply]
And I found it - Actiona. It clicks two times every 5 minutes on current time in the right menu of Juggluco, running in emulator, so graph is always actual. Thanks for tip) but auto-scroll feature is still very much needed to avoid making kludges)

**Comment by j-kaltes** [comment]
Juggluco 7.4.5 has a "_Now clamp_" option in settings.
https://www.juggluco.nl/Juggluco/download.html

**Reply by AgushkaNightscout** [reply]
Thank you very much! It works very well.
